addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1395,6 +1395,10 @@ const ColorpickerUserValueWidget = SelectUserValueWidget.extend({
             this._colorPaletteRenderPromise = this._renderColorPalette();
         }
         this._super(...arguments);
+        this.$el.find('[title]').tooltip({
```

Comment:
Can you move it before the super?
It is better to keep the super() either at the very top or at the very bottom so it is not "lost" between code blocks
But actually, since the `selector` attribut of the other `tooltip()` is acting as a jQuery `.on()`, it should (and is) initialize the tooltips on everything in that part of the DOM, even on dynamically injected DOM later (which is not the case for the colorpicker, it is just invisible in the DOM but already here?).

This shouldn't be needed at all, can you have a quick look why you need it?

For instance, if you just duplicate a node in the DOM through the browser debugger tools, and add a title attribute, you will see it automatically received the BS tooltip since it fits the `selector` attribute `[title]:not(.oe_snippet)`.
I'll let you check why is the colorpicker not behaving like that

Reply:
Indeed it's weird to need it. Here is what I understood:
what you say is true, by reading _/addons/web/static/lib/bootstrap/js/tooltip.js_      
I understood that doing this 
```js
this.$el.tooltip({
	selector: '[title]',
});
```
is (roughly) equivalent to doing this
```js
this.$el.on('mouseenter.bs.tooltip', '[title]', function (ev) {
	showTheTooltip();
 });
```
So it should work...
What I don't understand is if in this same place (ColorPickerUserValueWidget.open()) I put this
```js
this.$el.find('[title]').on('mouseenter', '', function (ev) {
          console.log('[FIND] mouseenter', ev.target.title);
});
```
I'll get some logs when I hover the elements of the color picker that have a title. But when I put this 
```js
this.$el.on('mouseenter', '[title]', function (ev) {
            console.log('[SELECTOR] mouseenter', ev.target.title);
});
```
the function given in parameter is never executed...
It seems to be a JQuerry problem, for which I don't think there is any better solution than to work around it...

Reply:
jQuery is rarely a problem (at least the kind of problem you described), it is quite battle tested as a library :)

The easiest way here to find what was the issue was quite simple, I'll share what I did to find out really quickly:
- First, I removed this `.tooltip()` diff
- As I said, first I edited the DOM and type `<p title="ok"/>` and saw that it was working as expected (which is expected because you do `this.$el.tooltip({ selector: 'we-title, [title]:not(.oe_snippet)',` on the other file, which will include everything (even dynamically injected content) in the $el.
- From there, I replicated the same step but added the `<p title="ok"/>` in the color picker element. Same, it was working as expected.
- There were only 2 differences between my `p` and the not working `<button class="w-50 o_custom_gradient_btn o_we_color_btn py-1 btn" title="Define a custom gradient" data-color="false">Custom</button>`.
First, the tag itself `p`vs `button`, second the attributes (`class` eg)
- I started to apply the classes to my `p` and suddenly it stopped working, without too much surprise.
- I removed one class at a time, ultimately figuring that the `o_we_color_btn` is preventing the tooltip to correctly appear.
- From there, I grepped the `o_we_color_btn` class in the code and saw a `'mouseenter .o_we_color_btn': '_onColorButtonEnter',` event
- The rest is pretty obvious, that method is doing a `ev.stopPropagation();` which just prevent the `.tooltip()` to act as it should. Same for the mouseleave, which also prevent the tooltip to close if you remove the stopPropagation from the mouseenter.

I am honestly not too sure why that `ev.stopPropagation();` is needed, it comes from [this quite old commit](https://github.com/odoo-dev/odoo/commit/22c5fc5dd2fcc7e37f103876a219727122d23293).
Maybe @qsm-odoo will know if those are useful or can simply be removed.